### PR TITLE
(rule):allow wildcard imports via editorconfig

### PR DESCRIPTION
**Pull request by aem**
while some people might not like wildcard imports, importing things like
an entire enum class via a wildcard is acceptable in most styleguides.
this PR adds an editorconfig option to override the default wildcard
functionality.

### CODE REVIEW

**Comment by aem** [comment]
this is related to #48 

while it's not up to this tool to be opinionated about IDEs, i think an override option is valid.

**Comment by jackylimel** [comment]
When will this pull request get merged? 

Android Studio cannot auto-import static functions, so wildcard import would be very convenient at this stage.

**Comment by shyiko** [comment]
@jackylimel 

imageedit_2_7341367885

peek 2017-08-28 05-46

**Comment by jackylimel** [comment]
It's very interesting that none of the Mockito static methods, like `when`, or `verify`, can be auto-imported..

**Comment by jackylimel** [comment]
but, anyway, JetBrains IDE can optimize imports on the fly, e.g. if there are many classes from the same package, IDE will auto convert the imports to wildcard import. It would be great to use the IDE feature, rather than import the classes one-by-one

**Comment by shyiko** [comment]
(unrelated to PR)

@jackylimel https://github.com/shyiko/ktlint#-with-intellij-idea contains info on how to turn 
> IDE will auto convert the imports to wildcard import

off (either manually or in completely automated fashion (`ktlint --apply-to-idea`)).

**Comment by jackylimel** [comment]
I still don't understand why you have a strong opinion again this PR.

**Comment by shyiko** [comment]
@jackylimel https://github.com/shyiko/ktlint/issues/48#issuecomment-312027052

**Comment by henrik242** [comment]
+1

File: ktlint-ruleset-standard/src/test/kotlin/com/github/shyiko/ktlint/ruleset/standard/NoWildcardImportsRuleTest.kt [hunk 1]
```diff
@@ -22,4 +22,17 @@ class NoWildcardImportsRuleTest {
...
+    fun testLintWithOverride() {
+        assertThat(NoWildcardImportsRule().lint(
+            """
+            import a.*
+            import a.b.c.*
+            import a.b
+            import kotlinx.android.synthetic.main.layout_name.*
+            """.trimIndent(),
+            mapOf("allow_wildcard_imports" to "true")
+        )).isEqualTo(listOf<LintError>())
```
Comment (jakubvimn) [inline ktlint-ruleset-standard/src/test/kotlin/com/github/shyiko/ktlint/ruleset/standard/NoWildcardImportsRuleTest.kt]: What about using "emptyList()"?

**Comment by sebastienrouif** [comment]
@aem could you catch up with master branch ? That would be very helpful to have an updated version of this branch and SHA1 to refer to in our local snapshot. I hope your branch gets merged one day

**Comment by aem** [comment]
@sebastienrouif sure, i'll resolve conflicts and update later today.

**Comment by aem** [comment]
@sebastienrouif @shyiko updated to latest master

**Comment by neworld** [comment]
Any news?

**Comment by aem** [comment]
@iTomkinas please see the discussion in #48. Simply put, it's the maintainer's decision about whether or not this is a valid inclusion in the linter, and if it is valid whether an editorconfig line is the right way to implement it. I made my proposed solution and it seems like it's not the approach the maintainer would choose themselves. I've left the PR open in case that decision is reversed, or in case someone would like to fork the project and include this patch on their own.

**Comment by aem** [comment]
it was decided this isn't a valid lint rule. closing this PR to indicate that the issue has been thoroughly discussed and reached its natural conclusion.

**Pull request closed without merging**